src/nvim/channel.h

```diff
@@ -53,6 +53,7 @@ typedef struct {
   garray_T buffer;
   bool eof;
   bool buffered;
+  char __pad[6];
```

Comment:
Ew, this is fugly. Shouldn't the compiler be able to do this stuff on is own?

Reply:
Yes it does exactly that. However, to remove the warnings introduced by `-Wpadded` the padding needs to be added manually.

**I don't plan to merge any of the manual padding.**

The goal here is to just re-arrange some fields in existing structs to use less memory.

E.g.

```c
struct {
    int a;
    long b;
    int c;
}
```

Uses 24 bytes in memory since the compiler will pad this to:

```c
struct {
    int a;
    char _pad0[4];
    long b;
    int c;
    char _pad1[4];
}
```

However, if we re-arrange the fields to:

```c
struct {
    int a;
    int c;
    long b;
}
```

This now uses only 16 bytes of memory (30% reduction) and has zero padding.

The goal of this PR is to find which structs have fields which can be re-arranged like this.

I'd like to add some checks to enforce this, but I can't find anything concrete.

Reply:
IIRC clang-tidy has a "could pad better" warning. Unsure if it's disabled at the moment, could take a look and see if we can use that.

Reply:
And thank you for the explanation.

Reply:
Oooh, that would be very helpful if it works.

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/altera/struct-pack-align.html

It's warns for both packing and aligning, unsure if we want both.

Reply:
Cool, how do I enable it?

Reply:
Add it to .clang-tidy file in the repo root

Reply:
Got it. It wasn't immediately obvious to me how the syntax of that file worked.